### PR TITLE
Heartbeat API

### PR DESCRIPTION
**Pull request by ErikBjare**
This is a work in progress, I've yet to test it with any watcher but feel free to comment on the code and submit improvements.

### TODO
 - [x] Tests
 - [x] Support in client library
 - [x] Actual use in a watcher (aw-watcher-window, or aw-watcher-web if I can throw together a prototype)

### CODE REVIEW

**Comment by ErikBjare** [comment]
Note: Needs the latest aw-core due to two changes I accidentally committed (one here and one there). See comment on diff.

File: aw_server/api.py [hunk 1]
```diff
@@ -185,15 +190,7 @@ def post(self, bucket_id):
...
-        events = []
-        if isinstance(data, dict):
-            events = [Event(**data)]
-        elif isinstance(data, list):
-            for e in data:
-                events.append(Event(**e))
-        else:
-            logger.error("Invalid JSON object")
-            raise BadRequest("InvalidJSON", "Invalid JSON object")
+        events = Event.from_json_obj(data)
```
Comment (ErikBjare) [inline aw_server/api.py]: This was what I referred to in my comment on the PR.

**Comment by ErikBjare** [comment]
There seems to be some issue with the pulsetime parameter. Will investigate another day.

**Comment by ErikBjare** [comment]
Completely forgot about this, I've been running activitywatch without issues for quite a while now (apart from an issue with aw-client). There are now plenty of watchers using the heartbeat API (aw-watcher-window, aw-watcher-web).

**Comment by johan-bjareholt** [comment]
aw-webui is still broken due to this change though, so it is not ready yet

**Comment by johan-bjareholt** [comment]
The only thing left in this PR that should be fixed is tests, and since heartbeat merge is now a transform it might be enough to just test that in aw_core since it's the only thing with high complexity.

So should we just merge this and add that test to aw_core?

**Comment by ErikBjare** [comment]
@johan-bjareholt Pretty sure such a test is already in `aw_core`.

I'm having some issues with the views right now, will investigate before merging.

**Comment by johan-bjareholt** [comment]
Oh, the coverage.io test failed in the aw-core PR so i assumed that there were no tests but now i see that they are.

**Comment by ErikBjare** [comment]
So, I guess the only thing remaining here is writing an integration test (`aw-server` + `aw-client`).

**Comment by ErikBjare** [comment]
Can't be bothered to write tests right now, it seems to work fine. Merging.

**Pull request merged**